### ISSUE TITLE
Issue with electricity and habitats upgrades.

### ISSUE THREAD

**Issue by Leo-13th** [heathcliff26/infraspace-savegame-editor#80]
Hi!
There is a issue when you select "Fill all habitats with workers". Electricity is 0/0 and cannot be changed by destroying or building any type of buildings.

Also i noticed that added workers have "_home": 0, instead of habitat ID. I guess that might cause some issues.

Another issue is that every time a change is made all habitats are upgrading themselves infinitly.
The only thing i found is that in a modified save file, line "upgrade" looks like:
"upgrade": "",
While in normal save file it is:
"upgrade": null,

I hope that helps and i hope you would manage to fix that.

**Comment by heathcliff26** [comment]
Hi,

i habe to admit i haven't played in a while so it might be behind. Could you provide me with the following:
1. Your Game Version
2. The unmodified save file
3. The modified save file

I assume the Problem only occurs when using "Fill all habitats with workers", or are you also experiencing trouble anywhere else?

**Comment by Leo-13th** [comment]
Hi,
Game version: 1.49.446
Saves (they are over 25MB :/ ):
https://file.io/Wf0MSmXntv4Z

The issue with endless upgrades of Lvl6 habitats occurs every time, no matter what you modify.
in original save it is:
"upgrade"= null, 
in modified it is:
"upgrade"= "",

The issue with 0/0 electricity occur only after selecting "Fill all habitats with workers".
I hope this helps.

**Comment by heathcliff26** [comment]
Thank you @Leo-13th for providing the data.
I could reproduce the issues on my end and created a series of PRs to fix them:
1. #82 fixes the habitats endlessly updating
2. #84 fixes the issue with the electricity grid
3. #83 Adds support for the latest game version, specifically the `upgradeManager`, which i see in the save but does not actually appear to be in the game yet?

This [CI run](https://github.com/heathcliff26/infraspace-savegame-editor/actions/runs/11628288073) should contain all the fixes. Could you check the resulting artifact to confirm that it fixes the issue on your side? If yes i can close the issue and create a release with the fixes.

**Comment by Leo-13th** [comment]
Everything works fine so far.
I will let you know if something will occur in the future.

**Comment by heathcliff26** [comment]
Yeah, i may not be actively playing InfraSpace right now, but if something breaks again and i have time i will fix it.
New version 1.1.2 should be released shortly.
